run_dim_reduction.py

```diff
@@ -65,16 +75,21 @@ def process_PCA(input_dir, output_dir, weights_dir, prefix, suffix='_after'):
         suffix (str): suffix for input file name
 
     """
-    model_path = os.path.join(weights_dir, 'pca_model.pkl')
+    # model_path = os.path.join(weights_dir, 'pca_model.pkl')
+    model_path = weights_dir
```

Comment:
`weights_dir` is supposed to be a directory where the model is saved. The pickle file name needs to be specified for the loading to work.

Reply:
In the test case for microglia (in the `bryant_rubella_experiments.yml` I do specify the weight file directly in the config file.  We can rename the config parameter to `weights_file` or approach this another way?

Reply:
The way I organize the output files is all the model files have the same name (pca_model.pkl). the model file and its related outputs (such as PCA plots) are saved in weight_dir which is named differently for different models. So you only need to specify weight_dir since the model name is always  'pca_model.pkl'. I found it's cleaner to organize the model and its related files this way. But we can also use weights_file if you want to have different names for the models

Reply:
The way `fit_PCA` is written in this repo is that all new PCA models get different names based on `strftime("%Y_%m_%d_%H%M", gmtime())`. They will all be placed in the same directory supplied by `config.dim_reduction.weights_dirs`.  So if you run the same fitting procedure multiple times, they will never overwrite old ones.

I think I like your approach though, which is to define a subfolder that can be experiment-specific.  Will revert to this.